main.go

```diff
@@ -170,6 +182,18 @@ func main() {
 		}
 		exitOnError(instaScaleController.SetupWithManager(context.Background(), mgr), "Error setting up InstaScale controller")
 	}
+	groups, err := kubeClient.DiscoveryClient.ServerGroups()
+	exitOnError(err, "Could not get server groups and resources")
+	// fmt.Println(resources)
+	for _, g := range groups.Groups {
```

Comment:
I'd suggest to create a separate util function to encapsulate the logic, e.g., something like:

```go
func HasAPIResourceForGVK(dc discovery.DiscoveryInterface, gvk schema.GroupVersionKind) (bool, error) {
	gv, kind := gvk.ToAPIVersionAndKind()
	if resources, err := dc.ServerResourcesForGroupVersion(gv); err != nil {
		if apierrors.IsNotFound(err) {
			return false, nil
		}
		return false, err
	} else {
		for _, res := range resources.APIResources {
			if res.Kind == kind {
				return true, nil
			}
		}
	}
	return false, nil
}
```

Reply:
Thanks @astefanutti